lua/telescope/builtin/git.lua

```diff
@@ -144,20 +144,18 @@ git.status = function(opts)
 end
 
 local set_opts_cwd = function(opts)
-  local is_git_dir = function(path)
-    vim.fn.system('cd ' .. path .. ' && git rev-parse HEAD >/dev/null 2>&1')
-    if vim.v.shell_error ~= 0 then
-      error(path .. ' is not a git directory')
-    end
-  end
-
   if opts.cwd then
     opts.cwd = vim.fn.expand(opts.cwd)
-    is_git_dir(opts.cwd)
   else
-    is_git_dir(vim.fn.expand('%:p:h'))
-    --- Find root of git directory and remove trailing newline characters
-    opts.cwd = vim.fn.systemlist("git rev-parse --show-toplevel")[1]
+    opts.cwd = vim.fn.expand('%:p:h')
+  end
+
+  -- Find root of git directory and remove trailing newline characters
+  opts.cwd = vim.fn.systemlist("git -C " .. opts.cwd .. " rev-parse --show-toplevel")[1]
+
+  -- opts.cwd contains the error message from git command if not in a git dir
+  if 1 ~= vim.fn.isdirectory(opts.cwd) then
+    error(opts.cwd)
```

Comment:
This results in a "not" good error message. For me
```
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Reply:
And checking for `vim.v.shell_error` is way better than isdirectory.
Could you keep the path around so we can get `error(path .. ' is not a git directory')` and if ret val is 0 than set opts.cwd?
Do you know what i mean

Reply:
ok, the new commit uses `vim.v.shell_error` now, I also kept path around so that the error message is better.